enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/RaftServer.java

```diff
@@ -132,10 +139,23 @@ protected void channelRead0( ChannelHandlerContext channelHandlerContext,
         {
             RaftMessages.RaftMessage message = storeIdAwareMessage.message();
             StoreId storeId = storeIdAwareMessage.storeId();
-            if ( messageHandler.validate( message, storeId ) )
+
+            if ( storeId.equals( localDatabase.storeId() ) )
             {
                 messageHandler.handle( message );
             }
+            else
+            {
+                if ( localDatabase.isEmpty() )
+                {
+                    raftStateMachine.notifyNeedFreshSnapshot( null, message::from );
```

Comment:
It doesn't seem to be used anywhere from what I can tell. I think we can get rid of that parameter.